consumer/pdata/common.go

```diff
@@ -260,6 +260,14 @@ func (am AttributeMap) InitFromMap(attrMap map[string]AttributeValue) AttributeM
 	return am
 }
 
+// InitEmptyWithCapacity constructs an empty AttributeMap with predefined slice capacity.
+func (am AttributeMap) InitEmptyWithCapacity(cap int) {
```

Comment:
Can we add this in a separate PR? There are more places where we need to update and use this.